### DISCUSSION TITLE
Why use Bref.sh

### DISCUSSION THREAD

**Discussion by west-side-solutions** [brefphp/bref#833] [General]
Hello, 
Can you explain what is the advantage to use bref for PHP application on aws , for a symfony project for exemple while elastic beanstalk offer a full integration.

Thanks for all !

**Comment by marcusirgens** [comment]
I believe the question you are asking is not really limited to bref. What you're essentially asking is "when should I choose serverless" or "when should I be using serverless functions".

- [When (and why) not to go serverless](https://www.serverless.com/blog/when-why-not-use-serverless)
- [What is Bref and serverless](https://bref.sh/docs/)
- [Why use serverless computing](https://www.cloudflare.com/learning/serverless/why-use-serverless/)

**Comment by mnapoli** [comment]
+1 on the links above. If you want a TL/DR version:

- you'll be able to scale more, and more easily
- you'll save time managing servers, containers and other low-level details
- you'll have a more robust infrastructure because you'll be managing less of it (i.e. you'll delegate some of the work to AWS)
- you'll be able to save on costs most of the time

Of course not everything is 100% magical, YMMV, but that's the idea of it.
